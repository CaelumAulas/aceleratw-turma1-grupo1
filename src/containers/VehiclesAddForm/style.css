.vehiclesAddForm--inputs {
  display: flex;
  flex-direction: column;
  margin-bottom: 50px;
}

.vehiclesAddForm--select {
  text-align: left;
  display: flex;
  flex-direction: column;
}
