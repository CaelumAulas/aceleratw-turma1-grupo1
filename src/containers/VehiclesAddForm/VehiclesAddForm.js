import React, { useState } from 'react';

import {
  Button,
  TextField,
  MenuItem,
  Container,
} from '@material-ui/core';

import './style.css';

const VehiclesAddForm = ({ toSend }) => {

  function createData(index, vehicle) {
    return { index, vehicle };
  }

  const brands = [
    createData(1, 'Corsa'),
    createData(2, 'Uno'),
    createData(3, 'Gol'),
    createData(4, 'Argo'),
    createData(5, 'Fiesta'),
  ];

  const [brand, setBrand] = useState("");
  const [model, setModel] = useState("");
  const [year, setYear] = useState("");
  const [value, setValue] = useState("");
  const [errors, setErrors] = useState({
    validator: {
      buttonEnabled: false,
      isValid: true,
      text: ""
    }
  });

  const handleMarcaSelectionChange = (event) => {
    setBrand(event.target.value);
  };

  const handleModeloChange = (event) => {
    setModel(event.target.value);
  };

  const handleYearChange = (event) => {
    setYear(event.target.value);
  };

  const handleValueChange = (event) => {
    setErrors({ validator: isValueValid(event.target.value) })
    setValue(event.target.value);
  };

  function isValueValid(value) {
    if (+ value <= 0) {
      return { buttonEnabled: false, isValid: false, text: 'Adicione um valor maior que 0' }
    }
    return { buttonEnabled: true, isValid: true, text: "" }
  }

  function submitForm(event) {
    event.preventDefault();
    const bag = {
      marca: brand,
      modelo: model,
      ano: year,
      valor: value
    }
    toSend(bag);
  }

  function goBack(event) {
    event.preventDefault();
    window.history.back();
  }

  return (
    <Container>
      <form variant="filled"
        autoComplete="off"
        className="vehiclesAddForm"
        data-testid="VehiclesAddForm"
        noValidate
        onSubmit={(event) => submitForm(event)}
      >
        <div className="vehiclesAddForm--select">
          <TextField
            data-testid="brand"
            fullWidth
            id="standard-basic"
            label="Marca"
            margin="normal"
            onChange={handleMarcaSelectionChange}
            select
            value={brand}
            variant="outlined"
          >
            <MenuItem>
              <em>Nenhum</em>
            </MenuItem>
            {brands.map((row) => {
              return <MenuItem value={row.index} key={row.index}>{row.vehicle}</MenuItem>;
            })}
          </TextField>
        </div>
        <div className="vehiclesAddForm--inputs">
          <TextField 
            data-testid="model"
            fullWidth
            id="standard-basic"
            label="Modelo"
            margin="normal"
            required
            value={model}
            variant="outlined"
            onChange={handleModeloChange}
          />
          <TextField 
            data-testid="year"
            fullWidth
            id="standard-basic"
            label="Ano"
            margin="normal"
            onChange={handleYearChange}
            required
            type="number"
            value={year}
            variant="outlined"
            />
          <TextField 
            data-testid="price"
            error={!errors.validator.isValid}
            fullWidth
            helperText={errors.validator.text}
            id="standard-basic"
            label="Valor"
            margin="normal"
            required
            type="number"
            value={value}
            variant="outlined"
            onChange={handleValueChange}
          />
        </div>
        <Button 
          color="primary"
          disabled={
            !brand ||
            !model ||
            !year ||
            !value ||
            !errors.validator.buttonEnabled
          }
          variant="contained"
          type="submit"
        >
          Cadastrar
        </Button>
        <Button 
          variant="contained" 
          color="secondary"
          onClick={goBack}
        >
          Cancelar
        </Button>
      </form>
    </Container>
  )
}

export default VehiclesAddForm
