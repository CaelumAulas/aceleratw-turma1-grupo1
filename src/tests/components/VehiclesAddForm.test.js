import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react'

import VehiclesAddForm, {handleMarcaSelectionChange} from '../../containers/VehiclesAddForm/VehiclesAddForm'


describe('Testando componentes de Adicionar VeÃ­culos', () => {
    it('existe campo para adicionar a marca ', () => {
        render (<VehiclesAddForm/>);
        const brand = screen.getByTestId("brand");
        expect(brand).toBeInTheDocument();
    });
    it('existe campo para adicionar o modelo ', () => {
        render (<VehiclesAddForm/>);
        const model = screen.getByTestId("model");
        expect(model).toBeInTheDocument();
    });
    it('existe campo para adicionar o ano ', () => {
        render (<VehiclesAddForm/>);
        const year = screen.getByTestId("year");
        expect(year).toBeInTheDocument();
    });
    it('existe campo para adicionar o valor ', () => {
        render (<VehiclesAddForm/>);
        const price = screen.getByTestId("price");
        expect(price).toBeInTheDocument();
    });
    it('valor menor que 0 retorna erro ', () => {
        render (<VehiclesAddForm/>);

        const price = screen.getByTestId( "price");
        price.value = 30;
      
        const message = isValueValid(price.value);

        expect(message.text).toBe("");
    });
})
