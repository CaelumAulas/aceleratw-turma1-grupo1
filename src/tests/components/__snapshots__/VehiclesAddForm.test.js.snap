// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando componentes de Adicionar Veículos valor menor que 0 retorna erro  1`] = `
<div
  class="MuiFormControl-root MuiTextField-root"
  data-testid="price"
>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
    data-shrink="false"
    for="price"
    id="price-label"
  >
    Valor
    <span
      aria-hidden="true"
      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
    >
       
      *
    </span>
  </label>
  <div
    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiInput-input"
      id="price"
      required=""
      type="number"
      value=""
    />
  </div>
  
</div>
`;
